sciris/sc_utils.py

```diff
@@ -1553,8 +1554,61 @@ def flattendict(inputdict=None, basekey=None, subkeys=None, complist=None, keyli
             complist.append(comp)
     return complist, keylist
 
+def suggest(user_input, valid_inputs, threshold=4, die=False):
+    """
+    Return suggested item
+
+    Returns item with lowest Levenshtein distance, where case substitution and stripping
+    whitespace are not included in the distance. If there are ties, then the additional operations
+    will be included
+
+    :param user_input: String with user's input
+    :param valid_inputs: List/collection of valid strings
+    :param threshold: Maximum number of edits required for an option to be suggested
+    :param die: If True, an informative error will be raised (to avoid having to implement this in the calling code)
+    :return: Suggested string. Returns None if no suggestions with edit distance less than threshold were found. This helps to make
+             suggestions more relevant.
 
+    Example usage:
+
+    >>> suggest('foo',['Foo','Bar'])
+    'Foo'
+    >>> suggest('foo',['FOO','Foo'])
+    'Foo'
+    >>> suggest('foo',['Foo ','boo'])
+    'Foo '
 
+    """
+
+    if isstring(valid_inputs):
+        valid_inputs = [valid_inputs]
+    else:
+        valid_inputs = list(valid_inputs)
```

Comment:
Could do `promotetolist()` :)

Reply:
Did that first, but it fails if you pass in a set because it just wraps the set in a list!

Reply:
That sounds like a bug in `promotetolist()` that should be fixed :)